nixos/tests/clevis.nix

```diff
@@ -0,0 +1,440 @@
+              };
+            };
+          ''}
+        };
+
+        # To ensure that we can rebuild the grub configuration on the nixos-rebuild
+        system.extraDependencies = with pkgs; [ stdenvNoCC ];
+        documentation.enable = false;
+        hardware.enableAllFirmware = lib.mkForce false;
+        networking.interfaces."eth1".useDHCP = false;
+        ${replaceStrings [ "\n" ] [ "\n  " ] extraConfig}
+      }
+    '';
+
+  testScriptFun = { createRootPartition, extraConfig, systemdStage1, fallback, preBootCommands }:
+    let
+      createQemuMachine = ''
+        import os
+
+        # Set the QEMU flags to create the machine where we install NixOS
+        def assemble_qemu_flags():
+            flags = "-cpu max"
+            ${if (system == "x86_64-linux" || system == "i686-linux") then ''
+              flags += " -m 1024"
+            '' else ''
+              flags += " -m 768 -enable-kvm -machine virt,gic-version=host"
+            ''}
+            flags += " -chardev socket,id=chrtpm,path=/tmp/swtpm-sock -tpmdev emulator,id=tpm0,chardev=chrtpm -device tpm-tis,tpmdev=tpm0"
+            flags += " -device virtio-net-pci,netdev=vlan1,mac=52:54:00:12:11:02 -netdev vde,id=vlan1,sock=\"$QEMU_VDE_SOCKET_1\""
+            return flags
+
+
+        qemu_flags = {"qemuFlags": assemble_qemu_flags()}
+
+        image_dir = installer.state_dir
+        disk_image = os.path.join(image_dir, "installer.qcow2")
+
+        hd_flags = {
+            "hdaInterface": "virtio",
+            "hda": disk_image,
+        }
+
+        default_flags = {**hd_flags, **qemu_flags}
+
+        def create_machine_named(name):
+            return create_machine({**default_flags, "name": name})
+      '';
+
+      tpmEmulator = ''
+        import subprocess
+
+        # From systemd-initrd-luks-tpm2.nix
+        class Tpm:
+            def __init__(self):
+                self.start()
+
+            def start(self):
+                self.proc = subprocess.Popen(["${pkgs.swtpm}/bin/swtpm",
+                    "socket",
+                    "--tpmstate", "dir=/tmp/swtpm",
+                    "--ctrl", "type=unixio,path=/tmp/swtpm-sock",
+                    "--tpm2"
+                    ])
+
+                # Check whether starting swtpm failed
+                try:
+                    exit_code = self.proc.wait(timeout=0.2)
+                    if exit_code is not None and exit_code != 0:
+                        raise Exception("failed to start swtpm")
+                except subprocess.TimeoutExpired:
+                    pass
+
+            """Check whether the swtpm process exited due to an error"""
+            def check(self):
+                exit_code = self.proc.poll()
+                if exit_code is not None and exit_code != 0:
+                    raise Exception("swtpm process died")
+      '';
+    in
+
+    if !pkgs.stdenv.hostPlatform.isx86 then ''
+      installer.succeed("true")
```

Comment:
If the test only works on x86, then it should have the appropriate `supportedPlatforms`, rather than being a no-op derivation. I know `installer.nix` does it this way, but it probably shouldn't.

Reply:
Not sure about what you're referring to by `supportedPlatforms`, seen no other test using that. I've instead changed `handleTest` by `handleTestOn [ "x86_64-linux"]` in `all-tests.nix`